src/frontend/src/scheduler/hummock_snapshot_manager.rs

```diff
@@ -34,7 +34,12 @@ const UNPIN_INTERVAL_SECS: u64 = 10;
 
 pub type HummockSnapshotManagerRef = Arc<HummockSnapshotManager>;
 pub enum PinnedHummockSnapshot {
-    FrontendPinned(HummockSnapshotGuard),
+    FrontendPinned(
+        HummockSnapshotGuard,
+        // `only_checkpoint_visible`.
+        // It's embedded here because we always use it together with snapshot.
+        bool,
```

Comment:
Refactor to embed `only_checkpoint_visible` here, instead of previously
- passing another boolean variable around. It complicates interfaces.
- reading from session config each time. It is unsafe because config can be changes between multiple reads within a same query.

@xxhZs